components/authz-service/server/v2/rules_property_test.go

```diff
@@ -18,17 +21,511 @@ import (
+		))
+
+	properties.Property("updates an applied and staged rule and applies update confirming transition from staged to applied",
+		prop.ForAll(
+			func(reqs projectAndRuleReq) bool {
+				defer testDB.Flush(t)
+
+				_, err := cl.CreateProject(ctx, &reqs.CreateProjectReq)
+				if err != nil {
+					return reportErrorAndYieldFalse(t, err)
+				}
+
+				_, err = cl.CreateRule(ctx, &reqs.rules[0])
+				if err != nil {
+					return reportErrorAndYieldFalse(t, err)
+				}
+
+				cl.ApplyRulesStart(ctx, &api.ApplyRulesStartReq{})
+
+				updateReq := api.UpdateRuleReq{
+					Id:         reqs.rules[0].Id,
+					ProjectId:  reqs.rules[0].ProjectId,
+					Name:       reqs.rules[0].Name + " updated",
+					Type:       reqs.rules[0].Type,
+					Conditions: reqs.rules[0].Conditions,
+				}
+				rInitialStaged, err := cl.UpdateRule(ctx, &updateReq)
+				if err != nil {
+					return reportErrorAndYieldFalse(t, err)
+				}
+
+				updateReq2 := api.UpdateRuleReq{
+					Id:         reqs.rules[0].Id,
+					ProjectId:  reqs.rules[0].ProjectId,
+					Name:       reqs.rules[0].Name + " updated again",
+					Type:       reqs.rules[0].Type,
+					Conditions: reqs.rules[0].Conditions,
+				}
+				rFinalStaged, err := cl.UpdateRule(ctx, &updateReq2)
+				if err != nil {
+					return reportErrorAndYieldFalse(t, err)
+				}
 
+				cl.ApplyRulesStart(ctx, &api.ApplyRulesStartReq{})
+
+				rApplied, err := cl.GetRule(ctx, &api.GetRuleReq{Id: reqs.rules[0].Id})
+				if err != nil {
+					return reportErrorAndYieldFalse(t, err)
+				}
+
+				return rInitialStaged.Rule.Status == "" && // TODO: this is wrong; should be "staged"
+					rFinalStaged.Rule.Status == "" && // TODO: this is wrong; should be "staged"
+					rApplied.Rule.Status == "applied" &&
+					ruleMatches(updateReq, *rInitialStaged.Rule) &&
+					ruleMatches(updateReq2, *rFinalStaged.Rule) &&
+					ruleMatches(updateReq2, *rApplied.Rule)
+			},
+			createProjectAndRuleGen,
+		))
+	properties.TestingRun(t)
+}
+
+func TestDeleteRuleProperties(t *testing.T) {
+	ctx := context.Background()
+	cl, testDB, _, _, seed := testhelpers.SetupProjectsAndRulesWithDB(t)
+	properties := getGopterParams(seed)
+	_, _, createProjectAndRuleGen := getGenerators()
+
+	properties.Property("deletes a staged rule and applies deletion confirming rule not found in both states",
+		prop.ForAll(
+			func(reqs projectAndRuleReq) bool {
+				defer testDB.Flush(t)
+
+				_, err := cl.CreateProject(ctx, &reqs.CreateProjectReq)
+				if err != nil {
+					return reportErrorAndYieldFalse(t, err)
+				}
+
+				_, err = cl.CreateRule(ctx, &reqs.rules[0])
+				if err != nil {
+					return reportErrorAndYieldFalse(t, err)
+				}
+
+				_, err = cl.DeleteRule(ctx, &api.DeleteRuleReq{Id: reqs.rules[0].Id})
+				if err != nil {
+					return reportErrorAndYieldFalse(t, err)
+				}
+
+				_, err = cl.GetRule(ctx, &api.GetRuleReq{Id: reqs.rules[0].Id})
+				deletedWhenStaged := err != nil && strings.Contains(err.Error(), "could not find")
```

Comment:
Could we not use `AssertCode` here? Instead of `strings.Contains()`?
```suggestion
				deletedWhenStaged := grpc_test.AssertCode(t, codes.NotFound, err)
```
or something like that? [AssertCode](https://github.com/chef/automate/blob/6f4f5c482549dc615a99f68d3f5e87626310b1c8/lib/grpc/grpctest/status.go#L39) checks if `err != nil`, too.